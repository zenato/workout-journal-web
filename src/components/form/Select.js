import React from 'react';
import classNames from 'classnames/bind';
import styles from './Select.scss';

const cx = classNames.bind(styles);

const Select = ({ className, ...props }) => (
  <select
    {...props}
    className={cx('form-control', className)}
  />
);

export default Select;
