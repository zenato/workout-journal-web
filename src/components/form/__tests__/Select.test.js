import React from 'react';
import { shallow } from 'enzyme';
import Select from '../Select';

it('renders error', () => {
  const wrapper = shallow(
    <Select name="test">
      <option value="1">option 1</option>
    </Select>
  );
  expect(wrapper).toMatchSnapshot();
});
