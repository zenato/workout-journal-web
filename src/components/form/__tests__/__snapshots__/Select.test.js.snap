// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders error 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <select
    className="form-control"
    name="test"
>
    <option
        value="1"
    >
        option 1
    </option>
</select>,
  "nodes": Array [
    <select
      className="form-control"
      name="test"
>
      <option
            value="1"
      >
            option 1
      </option>
</select>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Select
        name="test"
>
        <option
                value="1"
        >
                option 1
        </option>
</Select>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <option
            value="1"
>
            option 1
</option>,
          "name": "test",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <select
          className="form-control"
          name="test"
>
          <option
                    value="1"
          >
                    option 1
          </option>
</select>,
        "_debugID": 2,
        "_renderedOutput": <select
          className="form-control"
          name="test"
>
          <option
                    value="1"
          >
                    option 1
          </option>
</select>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Select
    name="test"
>
    <option
        value="1"
    >
        option 1
    </option>
</Select>,
}
`;
